8-application-demos/6-kalshi-bet-predictor/main.py

```diff
@@ -0,0 +1,203 @@
+            "Only output the rewritten factual question with no explanations or commentary.\n"
+            f"STATEMENT: {rules}"
+        )
+
+        raw_response = self._generate_response(prompt)
+
+        return raw_response
+
+    def get_relevant_questions(self, question: str) -> list[str]:
+
+        prompt = (
+            "Based on the following question, generate a list of 5 relevant questions "
+            "that one could search online to gather more information. "
+            "These questions should yield information that would be helpful to answering "
+            "the following question in an objective manner.\n\n"
+            "Your response SHOULD ONLY BE the following lines, in this exact format:\n"
+            "1. <question 1>\n"
+            "2. <question 2>\n"
+            "3. <question 3>\n"
+            "4. <question 4>\n"
+            "5. <question 5>\n"
+            "Do not add ANY preamble, conclusion, or extra text.\n\n"
+            f"Question: \"{question}\""
+        )
+
+        raw_response = self._generate_response(prompt)
+
+        relevant_questions = []
+        for line in raw_response.split('\n'):
+            line = line.strip()
+            if line and line[0].isdigit():
+                clean_question = line.split('.', 1)[-1].strip()
+                relevant_questions.append(clean_question)
+        
+        print(f"Generated relevant questions: {relevant_questions}")
+
+        return relevant_questions
+
+    
+    def get_information(self, questions):
+        results = [self.exa.answer(q, text=True) for q in questions]
+        answers = [r.answer for r in results]
+        return answers
+
+    def get_binary_answer_with_percentage(self, information: str, question: str) -> Tuple[str, str, str]:
+        prompt = (
+            "Analyze the provided information below to answer the given binary question. "
+            "Based on the information, determine the probability that the answer is 'Yes' or 'No'.\n\n"
+            "--- Information ---\n"
+            f"{information}\n\n"
+            "--- Question ---\n"
+            f"{question}\n\n"
+            "IMPORTANT INSTRUCTIONS:\n"
+            "1. Your response MUST ONLY be a single line in THIS EXACT FORMAT:\n"
+            "   Yes: <YES PERCENTAGE>%, No: <NO PERCENTAGE>%, Explanation: <EXPLANATION>\n"
+            "2. Percentages must sum to 100%.\n"
+            "3. Do NOT include any preamble, summary, or additional text.\n"
+            "4. Provide a brief but clear explanation supporting your probabilities.\n\n"
+        )
+
+        response = self._generate_response(prompt)
+
+        match = re.search(r"Yes: (.*?), No: (.*?), Explanation: (.*)", response, re.DOTALL)
+
+        if match:
+            yes, no, explanation = match.groups()
+            return yes.strip(), no.strip(), explanation.strip()
+        else:
+            raise ValueError(f"Failed to parse LLM response: {response}")
+    
+    def evaluate(self, question):
```

Comment:
For code readability, I would keep this function in this file, and move the functions inside here into a seperate file called helper.py or analyst.py etc. Then you can do small inline comments about what each function does in this function.

Reason: I don't have to read through all the above code unless I want to dig deeper and can get to the crux of what you are doing